src/components/Modal/index.tsx

```diff
@@ -28,12 +28,24 @@ export interface ModalProps extends StandardProps {
    * Event triggered when the modal should be closed by the user.
    */
   onClose?(e: ModalCloseEvent): void;
+  /**
+   * Show or hide close modal button
+   * @default true
+   */
+  showCloseButton?: boolean;
+  /**
+   * Determines if modal should be closed when clicked outside or Esc pressed
+   * @default true
+   */
+  closeOnOutsideClick?: boolean;
```

Comment:
Looks fine to me (however, same argument as above) even though I think it is redundant. Keep in mind that the `onClose` allows the user to handle if it should be closed. We have a background origin to indicate that the modal should be closed due to clicking outside. I would not introduce too many redundant properties (especially if they are less powerful / flexible).

Reply:
I think if Modal specification has onClose animation, this should be handled by modal itself and not from outside. In this case implementation seems to be pretty complicated but overall it's as simple as possible for this need and pretty logical. onClose handler called after all animations played and modal is ready to be deleted from DOM

Reply:
removed either